src/apps/config/leader.lua

```diff
@@ -741,25 +765,20 @@ function selftest ()
       app_graph.app(graph, "source", basic_apps.Source, {})
       app_graph.app(graph, "sink", basic_apps.Sink, {})
       app_graph.link(graph, "source.foo -> sink.bar")
-      return graph
+      return {graph}
    end
+   local worker_start_code = "require('apps.config.leader').test_worker()"
    app_graph.app(graph, "leader", Leader,
-                 {setup_fn=setup_fn, follower_pids={S.getpid()},
+                 {setup_fn=setup_fn, worker_start_code=worker_start_code,
                   -- Use a schema with no data nodes, just for
                   -- testing.
                   schema_name='ietf-inet-types', initial_configuration={}})
-   app_graph.app(graph, "follower", require('apps.config.follower').Follower,
-                 {})
    engine.configure(graph)
    engine.main({ duration = 0.05, report = {showapps=true,showlinks=true}})
-   assert(app.app_table.source)
-   assert(app.app_table.sink)
-   assert(app.link_table["source.foo -> sink.bar"])
+   assert(app.app_table.leader.followers[1])
```

Comment:
What's the `1` here for?  I would have thought that you'd get some other ID.  Maybe I am missing this :)

Reply:
The device id for the test leader app is `1`. This could be different if preferable. The `setup_fn` returns a table of `{id => app graph, ...}` in this case our `setup_fn` simply returns `{graph}` (https://github.com/Igalia/snabb/blob/c7b918c7d39c8758bd71e6b378c0fed797f50200/src/apps/config/leader.lua#L768) so lua will give it the ID of 1.

Reply:
Ah makes sense to me, tx for explanation.